src/main/java/seedu/address/storage/JsonFriendlyLinkStorage.java

```diff
@@ -12,42 +12,42 @@
 import seedu.address.commons.exceptions.IllegalValueException;
 import seedu.address.commons.util.FileUtil;
 import seedu.address.commons.util.JsonUtil;
-import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.ReadOnlyFriendlyLink;
 
 /**
- * A class to access AddressBook data stored as a json file on the hard disk.
+ * A class to access FriendlyLink data stored as a json file on the hard disk.
  */
-public class JsonAddressBookStorage implements AddressBookStorage {
+public class JsonFriendlyLinkStorage implements FriendlyLinkStorage {
 
-    private static final Logger logger = LogsCenter.getLogger(JsonAddressBookStorage.class);
+    private static final Logger logger = LogsCenter.getLogger(JsonFriendlyLinkStorage.class);
 
-    private Path filePath;
+    private final Path filePath;
 
-    public JsonAddressBookStorage(Path filePath) {
+    public JsonFriendlyLinkStorage(Path filePath) {
         this.filePath = filePath;
     }
 
-    public Path getAddressBookFilePath() {
+    public Path getFriendlyLinkFilePath() {
         return filePath;
     }
 
     @Override
-    public Optional<ReadOnlyAddressBook> readAddressBook() throws DataConversionException {
-        return readAddressBook(filePath);
+    public Optional<ReadOnlyFriendlyLink> readFriendlyLink() throws DataConversionException {
+        return readFriendlyLink(filePath);
     }
 
     /**
-     * Similar to {@link #readAddressBook()}.
+     * Similar to {@link #readFriendlyLink()}.
      *
      * @param filePath location of the data. Cannot be null.
      * @throws DataConversionException if the file is not in the correct format.
      */
-    public Optional<ReadOnlyAddressBook> readAddressBook(Path filePath) throws DataConversionException {
+    public Optional<ReadOnlyFriendlyLink> readFriendlyLink(Path filePath) throws DataConversionException {
         requireNonNull(filePath);
 
-        Optional<JsonSerializableAddressBook> jsonAddressBook = JsonUtil.readJsonFile(
-                filePath, JsonSerializableAddressBook.class);
-        if (!jsonAddressBook.isPresent()) {
+        Optional<JsonSerializableFriendlyLink> jsonAddressBook = JsonUtil.readJsonFile(
+                filePath, JsonSerializableFriendlyLink.class);
+        if (jsonAddressBook.isEmpty()) {
```

Comment:
I like this part where you changed not isPresent() to isEmpty(), which makes the code cleaner